src/components/Map/MainMap.js

```diff
@@ -21,37 +20,48 @@ class MainMap extends Component {
             SCENARIOS: [],
             scenario: '',         
             dateSliderActiveMap: false,
-            countyBoundaries: { "type": "FeatureCollection", "features": []},
+            countyBoundaries: {},
             statsForCounty: {},
             currentDateIndex: 0,
-            dataLoaded: false
+            dataLoaded: false,
+            isLoading: true
         };
     };
 
-    componentDidMount() {
+    async componentDidMount() {
         const { geoid, dataset } = this.props;
-        this.initializeMap(geoid, dataset)
+        const state = geoid.slice(0, 2);
+        
+        try {
+            this.setState({isLoading: true});
+            const statsForMap = await fetchJSON('statsForMap');
+            const countyBoundaries = await fetchJSON('countyBoundaries');
```

Comment:
We may want to start introducing redux soon. It will make fetching API calls much easier. It lets you decouple the API call and the results from an individual component. This allows any child component to have access to the results (and any other state you want to be global) as well as the API call that fetches the data.